apps/studio/src/features/editing-experience/components/RootStateDrawer.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  VStack,
+  HStack,
+  Box,
+  Divider,
+  Spacer,
+  Button,
+  Text,
+} from '@chakra-ui/react'
+import {
+  DropResult,
+  DragDropContext,
+  Droppable,
+  Draggable,
+} from '@hello-pangea/dnd'
+import { BsPlus } from 'react-icons/bs'
+import { MdOutlineDragIndicator } from 'react-icons/md'
+import { RootDrawerState } from './EditPageDrawer'
+import { useState } from 'react'
+
+export type Block = {
+  text: string
+  id: string
+}
+
+type RootStateDrawerProps = {
+  blocks: Block[]
+}
+export default function RootStateDrawer({ blocks }: RootStateDrawerProps) {
+  const [currState, setCurrState] = useState<{ blocks: Block[] }>({ blocks })
+
+  const onDragEnd = (result: DropResult) => {
+    if (!result.destination) return
+
+    const updatedBlocks = Array.from(currState.blocks)
+    // Remove block at source index
+    const [movedBlock] = updatedBlocks.splice(result.source.index, 1)
+    // Insert at destination index
+    updatedBlocks.splice(result.destination.index, 0, movedBlock!)
+
+    setCurrState({ blocks: updatedBlocks })
```

Comment:
did we take this from our existing code?

Reply:
what do you mean by existing code? like classic code? nope

Reply:
ok sure, could i just check these behaviours:
1. dropping outside of dropzone
2. happy path dnd
3. dropping and releasing into original position

Reply:
@seaerchin no logic for those has been added yet. this is purely UI for displaying page structure.

Currently, dropping outside dropzone does nothing. Same for point (3). 

Happy path is to drag and drop or rearrange the blocks and the preview on the right should update. but these are logic to be tackled later